### PR TITLE
Implement decorrelated jitter backoff between retries in BalancingHttpClient

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: http-client/src/main/java/com/proofpoint/http/client/balancing/BalancingHttpClient.java [hunk 1]
```diff
@@ -248,26 +283,40 @@ else if (t instanceof FailureStatusException) {
...
+                                        }
+                                        catch (RuntimeException e1) {
+                                            setException(e1);
+                                        }
+                                    }
                                 }
-                                catch (Exception e2) {
-                                    setException(e2);
+                                catch (Exception e) {
+                                    e.printStackTrace();
```
Comment (gwittel) [inline http-client/src/main/java/com/proofpoint/http/client/balancing/BalancingHttpClient.java]: print?

Comment (johngmyers) [inline http-client/src/main/java/com/proofpoint/http/client/balancing/BalancingHttpClient.java]: oops

Comment (johngmyers) [inline http-client/src/main/java/com/proofpoint/http/client/balancing/BalancingHttpClient.java]: Removed this catch block. Also removed public visibility of new interface/class.

**Comment by johngmyers** [comment]
Squashed the fixups and merged

**Pull request closed without merging**